Do not overwrite previous status

arl opened the pull request.

arl committed:
Save and restore status string length

arl committed:
Set *out-of-repo* status string before tmgb string

Because the *out-of-repo* status string is set before, we keep using the
old syntax to set its style to remain compatible with older versions of
tmux-gitbar.

As the tmux-gitbar status string is set after and because we actually
have a specific style string for it, we can use the new syntax to set
the tmux-gitbar status string and style at the same time.

Note:

-old syntax:

```
set-window-option status-left-style STYLE
set-window-option status-left STATUS
```
-new syntax:

```
set-window-option status-left "#[STYLE]STATUS"
```

arl committed:
Do not touch the string length anymore

arl committed:
Use one tmux call to reset/update the status

aurelien-rainone commented:
I've been personally using this branch for 10 days now and there aren't problems. Before merging it I need to fix the tests, as now the expected behavior is different. If there's already a tmux status string where tmux-gitbar would be showed, this is not overwritten anymore, but the git status is appended to it instead. That's exactly where test on travis is failing

arl committed:
Fix typo

arl committed:
shellcheck [SC2086]: double quote

arl committed:
shellcheck [SC2086]: double quote

arl committed:
Add expect helper: assert_not_on_screen

arl committed:
Fix integration test: test new behavior

Integration tests were not passing anymore since they were testing
the old behavior, i.e they were checking that the tmux status
string was overwritten by the git status in case the current
directory was a Git tree. Now the integration tests check that the
git status is showing at the right of (i.e appended to) the tmux
status defined for the tmux-gitbar location (left or right)
